### PR TITLE
new-linter: ireturn (checks for function return type)

### PR DESCRIPTION
**Pull request by butuzov**
This PR introduces a new linter `ireturn` that checks for function's returned types and adds the possibility to reject or allow specific types of interfaces.

https://github.com/butuzov/ireturn

TL/DR
_Accept Interfaces, Return Concrete Types_

### CODE REVIEW

File: .golangci.example.yml [hunk 1]
```diff
@@ -447,6 +447,27 @@ linters-settings:
       - pkg: knative.dev/serving/pkg/apis/(\w+)/(v[\w\d]+)
         alias: $1$2
 
+  ireturn:
+    # ireturn allows to use either `allow` or `reject` settings at the same time.
+    # both settings are a lists of the keywords (`empty` for interface{}`,
+    # `error` for errors, `stdlib` for standard library, `anon` for anonymouse
+    # interfaces ) and reglar expresssions matched to interfaces or packages.
```
Comment (ldez) [inline .golangci.example.yml:454]:
```suggestion
  ireturn:
    # ireturn allows using either `allow` or `reject` settings at the same time.
    # both settings are lists of the keywords (`empty` for interface{}`,
    # `error` for errors, `stdlib` for standard library, `anon` for anonymous
    # interfaces ) and regular expressions matched to interfaces or packages.
```

File: .golangci.example.yml [hunk 2]
```diff
@@ -447,6 +447,27 @@ linters-settings:
...
+
+    allow:
+    # default settings are allow to use errors, empty interfaces, anonymouse
+    # interfaces, and interfaces provided by standard library.
```
Comment (ldez) [inline .golangci.example.yml:458]:
```suggestion
    # default settings are allowed to use errors, empty interfaces, anonymous
    # interfaces, and interfaces provided by the standard library.
```

File: go.mod [hunk 3]
```diff
@@ -13,6 +13,7 @@ require (
 	github.com/ashanbrown/makezero v0.0.0-20210520155254-b6261585ddde
 	github.com/bkielbasa/cyclop v1.2.0
 	github.com/bombsimon/wsl/v3 v3.3.0
+	github.com/butuzov/ireturn v0.0.0-20210903145654-645c1cbdbffd
```
Comment (ldez) [inline go.mod:16]: Can you create a tag? Like that bot can handle the updates.
And it's a good practice.

File: pkg/lint/lintersdb/manager.go [hunk 4]
```diff
@@ -456,6 +458,12 @@ func (m Manager) GetAllSupportedLinterConfigs() []*linter.Config {
 			WithSince("v1.36.0").
 			WithPresets(linter.PresetStyle).
 			WithURL("https://github.com/esimonov/ifshort"),
+
+		linter.NewConfig(golinters.NewIreturn(ireturnCfg)).
+			WithSince("v1.43.0").
+			WithPresets(linter.PresetStyle).
+			WithURL("https://github.com/butuzov/ireturn"),
+
```
Comment (ldez) [inline pkg/lint/lintersdb/manager.go:466]: can you remove the extra empty lines and move the linter near the end of the list?

File: test/testdata/ireturn_allow.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// args: -Eireturn
+// config: linters-settings.ireturn.allow=["IreturnAllowDoer"]
```
Comment (ldez) [inline test/testdata/ireturn_allow.go:2]: I recommend using a config file instead of the annotation/CLI approach.

**Review by ldez — CHANGES_REQUESTED** [review]
in the `test/testdata/` files can you try to use at least one imported element (from stdlib)?

File: go.mod (hunk 3 above)
Comment (butuzov) [reply go.mod:16]: Sure, just I wanted to add one more thing before going 0.1.0.

**Comment by butuzov** [comment]
All issues from the previous CR are fixed, I will be glad if you can take a second look to spot any issues that left?

File: pkg/config/linters_settings.go [hunk 6]
```diff
@@ -186,6 +187,11 @@ type ExhaustiveStructSettings struct {
 	StructPatterns []string `mapstructure:"struct-patterns"`
 }
 
+type IreturnSettings struct {
+	Allow  []string `mapstructure:"allow"`
+	Reject []string `mapstructure:"return"`
```
Comment (ldez) [inline pkg/config/linters_settings.go:192]:
```suggestion
	Reject []string `mapstructure:"reject"`
```

**Review by ldez — CHANGES_REQUESTED** [review]

File: pkg/config/linters_settings.go (hunk 6 above)
Comment (butuzov) [reply pkg/config/linters_settings.go:192]: WoW. How can I miss it? Merci

**Review by ldez — APPROVED** [review]

**Pull request merged**